test/run-test.ps1

```diff
@@ -1,42 +1,89 @@
+[cmdletbinding()]
+param(
+    [Parameter(Mandatory=$true)]
+    [ValidateSet("win", "linux")]
+    [string]$Platform
+)
+
 Set-StrictMode -Version Latest
 $ErrorActionPreference="Stop"
 
 $dockerRepo="microsoft/dotnet-nightly"
+$dirSeparator = [IO.Path]::DirectorySeparatorChar
 $repoRoot = Split-Path -Parent $PSScriptRoot
 
-if ($env:DEBUGTEST -eq $null) {
-    $optionalDockerRunArgs="--rm"
+if ($Platform -eq "win") {
+    $imageOs = "nanoserver"
+    $tagSuffix = "-nanoserver"
+    $testScriptSuffix = "ps1"
+    $containerRoot = "C:\"
+    $sdkRunArg = "powershell"
 }
 else {
-    $optionalDockerRunArgs=""
+    $imageOs = "debian"
+    $tagSuffix = ""
+    $testScriptSuffix = "sh"
+    $containerRoot = "/"
+    $sdkRunArg = ""
 }
 
 pushd $repoRoot
 
 # Loop through each sdk Dockerfile in the repo and test the sdk and runtime images.
-Get-ChildItem -Recurse -Filter Dockerfile | where DirectoryName -like "*\nanoserver\sdk\*" | foreach {
-    $sdkTag = $_.DirectoryName.Replace($repoRoot, '').Replace("\nanoserver", '').TrimStart('\').Replace('\', '-') + "-nanoserver"
+Get-ChildItem -Recurse -Filter Dockerfile |
+    where DirectoryName -like "*${dirSeparator}${imageOs}${dirSeparator}sdk${dirSeparator}*" |
+    foreach {
+        $sdkTag = $_.DirectoryName.
+                Replace("$repoRoot$dirSeparator", '').
+                Replace("$dirSeparator$imageOs", '').
+                Replace($dirSeparator, '-') +
+            $tagSuffix
 
-    $fullSdkTag="$($dockerRepo):$sdkTag"
-    $baseTag="$fullSdkTag".Replace("-sdk", '').Replace("-msbuild", '').Replace("-projectjson", '').Replace("-nanoserver", '')
+        $fullSdkTag="${dockerRepo}:${sdkTag}"
+        $baseTag=$fullSdkTag.
+            TrimEnd($tagSuffix).
+            TrimEnd("-msbuild").
+            TrimEnd("-projectjson").
+            TrimEnd("-sdk")
 
-    $timeStamp = Get-Date -Format FileDateTime
-    $appName="app$timeStamp"
-    $appDir="${repoRoot}\.test-assets\${appName}"
+        $timeStamp = Get-Date -Format FileDateTime
+        $appName="app$timeStamp"
+        $appDir="$repoRoot$dirSeparator.test-assets$dirSeparator$appName"
 
-    New-Item $appDir -type directory | Out-Null
+        New-Item $appDir -type directory | Out-Null
 
-    Write-Host "----- Testing $fullSdkTag -----"
-    docker run -t $optionalDockerRunArgs -v "$($appDir):c:\$appName" -v "$repoRoot\test:c:\test" --name "sdk-test-$appName" --entrypoint powershell "$fullSdkTag" c:\test\create-run-publish-app.ps1 "c:\$appName" "${sdkTag}"
-    if (-NOT $?) {
-        throw  "Testing $full_sdk_tag failed"
-    }
+        Write-Host "----- Testing $fullSdkTag -----"
+        docker run -it --rm `
+            -v "${appDir}:${containerRoot}${appName}" `
+            -v "${repoRoot}${dirSeparator}test:${containerRoot}test" `
+            --name "sdk-test-$appName" `
+            $fullSdkTag $sdkRunArg "${containerRoot}test${dirSeparator}create-run-publish-app.${testScriptSuffix}" "${containerRoot}${appName}" $sdkTag
```

Comment:
It was moved to the $sdkRunArg.  This made it easier to handle because it is not needed for linux.